compiler/rustc_lint_defs/src/builtin.rs

```diff
@@ -5082,3 +5082,31 @@ declare_lint! {
     };
     crate_level_only
 }
+
+declare_lint! {
+    /// The `mixed_export_name_and_no_mangle` lint detects mixed usage of `export_name` and `no_mangle`
+    /// where `no_mangle` is not used by the compiler.
+    ///
+    /// ### Example
+    ///
+    /// ```rust,compile_fail
+    /// #![deny(mixed_export_name_and_no_mangle)]
+    ///
+    /// #[no_mangle]
+    /// #[export_name = "foo"]
+    /// pub fn bar() {}
+    ///
+    /// fn main() {}
+    /// ```
+    ///
+    /// {{produces}}
+    ///
+    /// ### Explanation
+    ///
+    /// The compiler will not use the `no_mangle` attribute when generating the symbol name for the function,
+    /// as the `export_name` attribute is used instead. This can lead to confusion and is unnecessary.
+    ///
+    pub MIXED_EXPORT_NAME_AND_NO_MANGLE,
```

Comment:
Every lint needs to be register to a lint pass, but since this lint emitted outside a lint pass you can add it to the `HardwiredLints` pass at the top of this file:
https://github.com/rust-lang/rust/blob/77348fadcbd3a549556c51f64b4f8f19545e8f14/compiler/rustc_lint_defs/src/builtin.rs#L14-L18

Reply:
thanks for pointing out, added it together with the other comments, and all started to work out.